### ISSUE TITLE
Can't set memory (RAM)

### ISSUE THREAD

**Issue by TekkertheChaot** [GameServerManagers/LinuxGSM#1972]
**Game:** Minecraft
**LGSM install date:** 25 June 2018

# Describe the bug / occurrance
Changing the "javaram" parameter (either in `common.cfg` or `mcserver.cfg`) doesn't change the memory for the mcserver.

Content of `common.cfg`:
```
##################################
######## Common Settings #########
##################################
# PLACE GLOBAL SETTINGS HERE
## These settings will apply to all instances
javaram="4096"
```

Content of `mcserver.cfg`:
```
##################################
######## Instance Settings ########
##################################
# PLACE INSTANCE SETTINGS HERE
## These settings will apply to a specific instance
javaram="4096"
```

Output while starting server:
```
Command-line Parameters
===============================================
java -Xmx1024M -jar /home/mycraft/serverfiles/minecraft_server.jar nogui
```

# Solution (I came up with)
**TL;DR:** avoid those "bugs" by **[reading the damn manual!](https://github.com/GameServerManagers/LinuxGSM/wiki/LinuxGSM-Config#how-to-use)**
**Breakdown:** 
1. Copy the contents of `_default.cfg` into your `instance.cfg` or `common.cfg`.
2. Cut down the file to the only lines you need. In this case it's:
   - Parameter / Variables 
[`javaram="YOUR_RAM_AMOUNT_IN_MB"`]
   - Execute String 
[`executable="java -Xmx${javaram}M -jar ${serverfiles}/minecraft_server.jar"`]
3. **Important!** Check for empty lines (line breaks), because they **will** [throw some errors at LGSM startup](https://i.imgur.com/UisbUFi.png) and **can** cause some variable deformation like [this one](https://i.imgur.com/pQerSWz.png) which can lead to some unexpected behaviour from LGSM.

What's left should be something like this:
```
##################################
######## Instance Settings ########
##################################
# PLACE INSTANCE SETTINGS HERE
## These settings will apply to a specific instance
javaram="4096" # -Xmx$4096M
executable="java -Xmx${javaram}M -jar ${serverfiles}/minecraft_server.jar"
```
With that I was able to start my server with my desired memory! ^^
Output:
```
Command-line Parameters
================================================
java -Xmx4096M -jar /home/mycraft/serverfiles/minecraft_server.jar nogui
```
**Thanks to @UltimateByte for pointing out my mistake and @dgibbs64 for general help!**

---
**Edit:** better formatting
**Edit 2:** added solution to this post for better placement, added link to corresponding wiki-page

**Comment by UltimateByte** [comment]
This works fine for me.
I had poeple yelling about this already. Issue is that they don't put the variable where this is actually used, which is at the bottom of the config file...

**Comment by dgibbs64** [comment]
looks like it might need some clarity then. Not sure how to best achieve that currently

**Comment by TekkertheChaot** [comment]
> This doesn't happen when you copy the whole config file to your instance like you're supposed to.

Ok, you got me. I missed that part. Thank you!
I am spoilt from the Source-Gameserver configs... 

**Aftermath**
So i pasted the contents from `_default.cfg` into `mcserver.cfg` and removed **only** the alert stuff (that I don't currently need).
It now looks like [this (hastebin.com)](https://hastebin.com/raw/irolurodiw).
I think I missed something again, because I still don't get my memory assigned correctly, see output:
```
Command-line Parameters
===============================================================================================================================
java -Xmx1024M -jar /home/mycraft/serverfiles/minecraft_server.jar nogui
```

In addition to this, LGSM now reports non-executable code:
```/home/mycraft/lgsm/config-lgsm/mcserver/mcserver.cfg: line 6: $'\r': command not found
/home/mycraft/lgsm/config-lgsm/mcserver/mcserver.cfg: line 8: $'\r': command not found
/home/mycraft/lgsm/config-lgsm/mcserver/mcserver.cfg: line 11: $'\r': command not found
/home/mycraft/lgsm/config-lgsm/mcserver/mcserver.cfg: line 13: syntax error near unexpected token `$'{\r''
'home/mycraft/lgsm/config-lgsm/mcserver/mcserver.cfg: line 13: `fn_parms(){
```
I get that empty lines (breaks) are causing the `line XX: $'\r': command not found` error.
But the last one seems odd.

What I've missed?

**Comment by TekkertheChaot** [comment]
@dgibbs64 
> looks like it might need some clarity then. Not sure how to best achieve that currently

To be fair, I didn't properly read the Wiki page about the configuration so that fault was definitely on my side.

**Comment by TekkertheChaot** [comment]
I found a solution, added it to the main post and I'm happy now!
This issue/bug can be considered as solved!

**Thanks again @UltimateByte for pointing the issue out.**

**Comment by UltimateByte** [comment]
Added warning in the wiki:
https://github.com/GameServerManagers/LinuxGSM/wiki/LinuxGSM-Config#how-to-use

**Comment by Beguiled** [comment]
As an FYI for those who end up here via searches, the documentation has moved and is now found here:

https://docs.linuxgsm.com/configuration/linuxgsm-config
